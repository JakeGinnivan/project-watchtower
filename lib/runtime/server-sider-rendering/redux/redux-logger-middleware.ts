import * as redux from 'redux'

// @TODO revisit and make logger and action types correct
const reduxLoggerMiddleware = (logger: any) =>
    <S>() =>
        (next: redux.Dispatch<S>) =>
            (action: any) => {
                let actionDescription = action.type
                if (action.meta) {
                    actionDescription = `${actionDescription} > ${JSON.stringify(action.meta)}`
                }
                if (action.payload) {
                    const output = typeof action.payload === 'string' ? action.payload : '{...}'
                    // tslint:disable-next-line max-line-length
                    actionDescription = `${actionDescription} > ${action.payload ? `/w payload: ${output}` : ''}`
                }

                // new error payload strategy
                if (action.isError) {
                    const logObj = { err: action, event: 'ActionFailure' }
                    const logMsg = `Failed to perform client action ${actionDescription}`
                    logger.warn(logObj, logMsg)
                } else if (action.error) { // old error payload strategy
                    const logObj = { err: action, event: 'ActionFailure' }
                    const logMsg = `Failed to perform client action ${actionDescription}`

                    logger.warn(logObj, logMsg)
                }

                logger.debug(`dispatching: ${actionDescription}`)

                const result = next(action)

                logger.debug(`finished processing: ${actionDescription} (including react updates)`)

                return result
            }

export default reduxLoggerMiddleware
