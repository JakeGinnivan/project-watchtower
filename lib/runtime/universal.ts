/**
 * Watchtower has code which is unsafe for the client bundles.
 *
 * This universal file should always be safe to import from the client or server
 */

export * from './util/promise-tracker'
export * from './util/promise-completion-source'
export * from './util/function-timer'
export { WatchtowerBrowserRouter } from './server/ssr/watchtower-browser-router'
