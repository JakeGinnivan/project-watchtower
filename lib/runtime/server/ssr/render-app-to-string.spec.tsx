/**
 * @jest-environment node
 */

import * as React from 'react'
import * as redux from 'redux'
import { Route, Switch } from 'react-router-dom'
import renderToString from './render-app-to-string'
import { createConsoleLogger } from '../../universal'

const Home: React.SFC<{}> = () => <div>Home</div>
const TestApp: React.SFC<{}> = () => (
    <Switch>
        <Route exact path="/" component={Home} />
    </Switch>
)

it('renders app callback with router', () => {
    const store = redux.createStore(() => ({}))

    const renderResult = renderToString('/', store, createConsoleLogger(), _ => <TestApp />)

    expect(renderResult.renderMarkup).toMatchSnapshot()
})
