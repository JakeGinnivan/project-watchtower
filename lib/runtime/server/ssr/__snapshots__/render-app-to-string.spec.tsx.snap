// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app callback with router 1`] = `
Object {
  "css": "",
  "html": "<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1178726173\\">Home</div>",
  "ids": Array [],
  "rules": Array [],
}
`;
